src/logging.cc

```diff
@@ -1148,13 +1221,50 @@ void LogFileObject::Write(bool force_flush,
   if (base_filename_selected_ && base_filename_.empty()) {
     return;
   }
-
-  if (file_length_ >> 20U >= MaxLogSize() || PidHasChanged()) {
+  struct ::tm tm_time;
+  bool is_split = false;
+  if ("day" == FLAGS_log_rolling_policy) {
+    localtime_r(&timestamp, &tm_time);
+    if (tm_time.tm_year != tm_time_.tm_year
+        || tm_time.tm_mon != tm_time_.tm_mon
+        || tm_time.tm_mday != tm_time_.tm_mday) {
+      is_split = true;
+    }
+  } else if ("hour" == FLAGS_log_rolling_policy) {
+    localtime_r(&timestamp, &tm_time);
+    if (tm_time.tm_year != tm_time_.tm_year
+        || tm_time.tm_mon != tm_time_.tm_mon
+        || tm_time.tm_mday != tm_time_.tm_mday
+        || tm_time.tm_hour != tm_time_.tm_hour) {
+      is_split = true;
+    }
+  } else if (file_length_ >> 20U >= MaxLogSize() || PidHasChanged()) {
+    is_split = true;
+  }
+  if (is_split) {
     if (file_ != NULL) fclose(file_);
     file_ = NULL;
     file_length_ = bytes_since_flush_ = dropped_mem_length_ = 0;
     rollover_attempt_ = kRolloverAttemptFrequency - 1;
   }
+  if ((file_ == NULL) && (!inited_) && ("size" == FLAGS_log_rolling_policy)) {
+    CheckHistoryFileNum();
+    const char *filename = file_list_.back().name.c_str();
+    int fd = open(filename, O_WRONLY | O_CREAT | O_APPEND, FLAGS_logfile_mode);
+    if (fd != -1) {
+#ifdef HAVE_FCNTL
+      // Mark the file close-on-exec. We don't really care if this fails
+      fcntl(fd, F_SETFD, FD_CLOEXEC);
+#endif
+      file_ = fopen(filename, "a+"); // Read and append a FILE*.
+      if (file_ == NULL) {      // Man, we're screwed!, try to create new log file
```

Comment:
When's/where's the new log file created as mentioned in the comment? This is not clear.

Reply:
done